### PR TITLE
add accesskey description

### PR DESCRIPTION
**Pull request by OfekAvergil**
## Related Issues

Related https://github.com/descope/etc/issues/6938

## Description

Adding the description field to accessKey (create and update).
## Must

- [X] Tests
- [X] Documentation (if applicable)

### CODE REVIEW

File: descope/sdk/mgmt.go [hunk 1]
```diff
@@ -413,7 +413,7 @@ type AccessKey interface {
 	//
 	// The permittedIPs parameter is an optional list of IP addresses or CIDR ranges that are allowed to use this access key.
 	// If not provided, all IPs are allowed.
-	Create(ctx context.Context, name string, expireTime int64, roles []string, keyTenants []*descope.AssociatedTenant, userID string, customClaims map[string]any, permittedIPs []string) (string, *descope.AccessKeyResponse, error)
+	Create(ctx context.Context, name string, expireTime int64, roles []string, keyTenants []*descope.AssociatedTenant, userID string, customClaims map[string]any, description string, permittedIPs []string) (string, *descope.AccessKeyResponse, error)
```
Comment (nmacianx) [inline descope/sdk/mgmt.go:416]: please also add `description` to the comment above the function

File: descope/sdk/mgmt.go [hunk 2]
```diff
@@ -432,7 +432,7 @@ type AccessKey interface {
 	//
 	// IMPORTANT: All parameters will override whatever values are currently set
 	// in the existing access key. Use carefully.
-	Update(ctx context.Context, id, name string) (*descope.AccessKeyResponse, error)
+	Update(ctx context.Context, id, name string, description string) (*descope.AccessKeyResponse, error)
```
Comment (nmacianx) [inline descope/sdk/mgmt.go:435]: edit the comment since it says `Only the name is settable for the time being.`

**Review by nmacianx — APPROVED** [review]
LGTM Ofek! Great work 🙌

File: descope/sdk/mgmt.go [hunk 3]
```diff
@@ -410,7 +410,7 @@ type AccessKey interface {
 	// If userID is supplied, then authorization would be ignored, and access key would be bound to the users authorization
 	// The customClaims parameter is an optional map of claims and their values, and if it's provided then
 	// those claims will be present in the JWT returned by calls to ExchangeAccessKey.
-	//
+	// The description parameter is an optional text providing a brief summary about the accessKey.
```
Comment (nmacianx) [inline descope/sdk/mgmt.go:413]: typo: "access key" instead of "accessKey"

**Review by nmacianx — APPROVED** [review]

File: descope/internal/mgmt/accesskey.go [hunk 4]
```diff
@@ -47,14 +47,17 @@ func (a *accessKey) SearchAll(ctx context.Context, tenantIDs []string) ([]*desco
...
 	if id == "" {
 		return nil, utils.NewInvalidArgumentError("id")
 	}
 	if name == "" {
 		return nil, utils.NewInvalidArgumentError("name")
 	}
-	body := map[string]any{"id": id, "name": name, "description": description}
+	body := map[string]any{"id": id, "name": name}
+	if description != nil {
+		body["description"] = description
```
Comment (nmacianx) [inline descope/internal/mgmt/accesskey.go:59]: shouldn't it be `*description` here? since yoy want the string value of the description

File: descope/internal/mgmt/accesskey.go [hunk 5]
```diff
@@ -89,17 +92,20 @@ func (a *accessKey) Delete(ctx context.Context, id string) error {
...
 		"keyTenants":   makeAssociatedTenantList(tenants),
 		"userId":       userID,
 		"customClaims": customClaims,
-		"description":  description,
 		"permittedIps": permittedIPs,
 	}
+	if description != nil {
+		res["description"] = description
+	}
+	return res
```
Comment (nmacianx) [inline descope/internal/mgmt/accesskey.go:108]: i don't think it makes sense to use a pointer string in the creation since if not provided, an empty description will be used, right?

File: descope/internal/mgmt/accesskey_test.go [hunk 6]
```diff
@@ -129,23 +131,23 @@ func TestAccessKeyUpdateSuccess(t *testing.T) {
 		require.NoError(t, helpers.ReadBody(r, &req))
 		require.Equal(t, "ak1", req["id"])
 		require.Equal(t, "abc", req["name"])
-		require.Equal(t, "desc", req["description"])
+		require.Equal(t, desc, req["description"])
 	}, response))
-	res, err := mgmt.AccessKey().Update(context.Background(), "ak1", "abc", "desc")
+	res, err := mgmt.AccessKey().Update(context.Background(), "ak1", "abc", &desc)
```
Comment (nmacianx) [inline descope/internal/mgmt/accesskey_test.go:136]: you're missing a test where you pass a nil description and check that the `description` is not in the sent request

**Review by nmacianx — COMMENTED** [review]
Commented :)

**Review by nmacianx — APPROVED** [review]
